src/benchcab/model.py

```diff
@@ -82,6 +82,36 @@ def model_id(self) -> int:
     def model_id(self, value: int):
         self._model_id = value
 
+    def get_coverage_dir(self) -> Path:
+        """Get absolute path for code coverage analysis."""
+        return (internal.CODECOV_DIR / f"R{self.model_id}").absolute()
+
+    def get_build_flags(self, mpi: bool, build_option: str, compiler_id: str) -> dict:
+        """Get flags for CMake build."""
+        build_flags = {}
+
+        build_flags["build_type"] = "Debug" if "debug" in build_option else "Release"
+        build_flags["mpi"] = "ON" if mpi else "OFF"
+
+        build_flags["flags_init"] = ""
+        if "codecov" in build_option:
+            if compiler_id != "ifort":
+                msg = f"""For code coverage, the only supported compiler is `ifort`
+                User has {compiler_id} in their environment"""
+                raise RuntimeError(msg)
```

Comment:
Yes, that is true, although it was decided that it would be clearer if `benchcab` gave the error message before passing in to the compiler (I can also add check for `ifx`). @ccarouge thoughts?